astropy/nddata/nddata.py

```diff
@@ -307,10 +312,42 @@ def __str__(self):
 
     def __repr__(self):
         prefix = self.__class__.__name__ + "("
-        data = np.array2string(self.data, separator=", ", prefix=prefix)
-        unit = f", unit='{self.unit}'" if self.unit is not None else ""
+        # to support reprs for other non-ndarray `data` attributes,
+        # add more cases here:
+        is_dask = da is not None and isinstance(self.data, da.Array)
 
-        return f"{prefix}{data}{unit})"
+        if (
+            isinstance(self.data, (int, float, np.ndarray))
+            or np.issubdtype(float, self.data)
+            or np.issubdtype(int, self.data)
+        ) and not is_dask:
+            # if data is an ndarray, get build a repr via Masked:
+            ma = Masked(self.data, mask=self.mask)
+            data_repr = repr(ma)
+            # strip the class name generated by Masked, and the final ")",
+            # and re-pad multi-line array outputs
+            after_first_paren = data_repr.index("(") + 1
+            old_prefix_spaces = " " * after_first_paren
+            new_prefix_spaces = " " * len(prefix)
+            data_repr = data_repr[after_first_paren:-1].replace(
+                old_prefix_spaces, new_prefix_spaces
+            )
+            unit = f", unit='{self.unit}'" if self.unit is not None else ""
+            return f"{prefix}{data_repr}{unit})"
+        else:
+            # relying on the builtin repr for objects other than ndarrays
+            # allows for compatibility with e.g. dask:
+            contents = []
+            for attr in ["data", "mask", "uncertainty", "unit"]:
```

Comment:
Nitpick. I don't think there is a difference but I prefer tuple in principle. Take it or leave it.

```suggestion
            for attr in ("data", "mask", "uncertainty", "unit"):
```